nemo/collections/asr/models/rnnt_models.py

```diff
@@ -660,6 +661,16 @@ def forward(
         return encoded, encoded_len
 
     # PTL-specific methods
+    def on_train_epoch_start(self) -> None:
```

Comment:
This is not good. Very easy to forget to add this in so many model classes (hybrid for example, ASR TTS for another). 

How about instead checking the decoding training type inside of greedy decoding itself and use that to decide whether to turn off or on cuda graphs.

Reply:
https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py#L372

Reply:
At first I thought this is an excellent place to toggle CUDA graphs.
But after reviewing the changes, I think it's better to use explicit `on_train_epoch_start`:
- disabling CUDA graphs in `on_train_epoch_start` is better, since this operation frees up the memory. This is critical for training, and the amount of memory can be large, especially if the alignments are preserved. This is impossible to do in the decoder when the training starts, but only after some training steps are done and we request decoding for the batch.
- hybrid CTC-RNNT is not a problem since it inherits `EncDecRNNTModel` (same for `EncDecRNNTBPEModel`)
- I fixed the ASR-TTS model. Thanks for catching the bug!

Reply:
> fixed the ASR-TTS model. Thanks for catching the bug!

See this is exactly the problem. I or future reviewers won't always remember every possible model that can be affected by this. 

Lets either put this inside the place i mentioned or put this in ASR Model and make it generic to all ASR models. Doing it per class is not future proof.